### ISSUE TITLE
Allow optimistic concurrency check on write to be optional

### ISSUE THREAD

**Issue by slashdotdash** [slashdotdash/eventstore#31]
Appending events to a stream requires an expected version. This is used for optimistic concurrency checking to ensure writers have not missed an event appended by another process. 

```elixir
:ok = EventStore.append_to_stream(stream_uuid, expected_version, events)
```

This check should be made optional to allow writers to append events without having first read them from the stream.

### Proposal

The `expected_version` argument should be extended to allow the following options:

- `:any_version` - no concurrency checking; allow any stream version (including no stream).
- `:no_stream` - ensure the stream does not exist.
- `:stream_exists` - ensure the stream exists.
- an integer  value - optimistic concurrency check against expected version (current behaviour).

#### Example usage

```elixir
:ok = EventStore.append_to_stream(stream_uuid, :any_version, events)
```

**Comment by HarenBroog** [comment]
@slashdotdash are there any plans of delivering this feature?

**Comment by slashdotdash** [comment]
@HarenBroog Yes, I've just done it! Published to Hex as [v0.12.0](https://hex.pm/packages/eventstore).
